dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/api/Meta.java

```diff
@@ -179,4 +203,16 @@ public String dataset(@PathParam("datasetId") Long datasetId, @QueryParam("exclu
         return retValue;
     }
 
+    private boolean isRestrictedByEmbargo(Dataset ds) {
```

Comment:
Hmm there is duplicated code in Access api, maybe some kind of embargo permission service should be created to check stuff about embargo in api.

Reply:
Ideally we would avoid as much logic as possible in api, since api should only be for routing traffic.

Reply:
Removing logic from `Access` api is not in scope of this task, and since `Meta` is deprecated and we have just two methods duplicated, I don't think its worth moving just this duplicated logic from both classes. 
That might change when we decide to move logic out from `Access` as a whole.

Reply:
I agree that moving logic from Access is not what you should do now. But since new code is made it can be done proper way without much pain since those methods would just be in other classe. Also I didn't mean only just this one method but all of those new methods you wrote.